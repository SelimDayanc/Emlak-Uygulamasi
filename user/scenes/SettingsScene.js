/**
 * @flow
 */
import React, {Component, PropTypes} from 'react';
import {ScrollView, StyleSheet, View} from 'react-native';
import SettingListItem from '../components/SettingListItem';
import EditProfile from '../components/profile/EditProfile';
import Separator from '../../components/Separator';
import {CountryPropType} from '../../property/common/proptypes';

export default class SettingsScene extends Component {
  static propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
    user: PropTypes.oneOfType([PropTypes.object, PropTypes.null]).isRequired,
    country: CountryPropType.isRequired,
    loadScene: PropTypes.func.isRequired,
  };

  render() {
    const {isAuthenticated, user, country, loadScene} = this.props;
    return (
      <ScrollView style={styles.container}>

        {isAuthenticated && <EditProfile loadScene={loadScene} user={user} />}

        <SettingListItem
          title="Upload Property"
          route="propertyCreate"
          loadScene={loadScene}
          icon="plus-square-o"
        />

        <Separator />

        <SettingListItem
          title="Change Country"
          route="countrySelect"
          loadScene={loadScene}
          icon="globe"
          selected={country.fullName}
        />

        <Separator />

        {isAuthenticated
          ? <View>
              <SettingListItem
                title="My Properties"
                route="manageProperties"
                loadScene={loadScene}
                icon="key"
              />
              <Separator />
              <SettingListItem
                title="Logout"
                route="logout"
                loadScene={loadScene}
                icon="key"
              />
            </View>
          : <SettingListItem
              title="Login"
              route="login"
              loadScene={loadScene}
              icon="key"
            />}
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    padding: 20,
  },
});
