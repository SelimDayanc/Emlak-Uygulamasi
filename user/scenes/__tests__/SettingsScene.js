import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import SettingsScene from '../../scenes/SettingsScene';

test('Settings Scene', () => {
  const tree = renderer
    .create(
      <SettingsScene
        isAuthenticated={true}
        user={{name: 'zal', email: 'z4ls@live.com'}}
        country={{
          fullName: 'Kuwait',
          abbr: 'KW',
          currency: 'KD',
          coords: {
            latitude: 29.3759,
            longitude: 47.9774,
          },
        }}
        loadScene={() => {}}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
