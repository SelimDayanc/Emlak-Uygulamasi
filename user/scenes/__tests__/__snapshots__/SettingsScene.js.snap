// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Scene 1`] = `
<RCTScrollView
  alwaysBounceHorizontal={undefined}
  alwaysBounceVertical={true}
  onContentSizeChange={null}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={undefined}
  onResponderTerminationRequest={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  onScrollShouldSetResponder={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  sendMomentumEvents={false}
  style={
    Array [
      Object {
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "overflow": "scroll",
      },
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "padding": 20,
      },
    ]
  }
>
  <RCTScrollContentView
    collapsable={false}
    removeClippedSubviews={undefined}
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "marginBottom": 50,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 2,
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#5d5f63",
                "fontSize": 20,
                "fontWeight": "700",
                "paddingVertical": 5,
              }
            }
          >
            zal
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#999",
                "fontWeight": "100",
                "paddingBottom": 5,
              }
            }
          >
            View and edit profile
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#5d5f63",
                  "fontSize": 80,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "flex": 1,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingVertical": 20,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 9,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#484848",
                "fontWeight": "200",
              }
            }
          >
            Upload Property
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#484848",
                  "fontSize": 20,
                },
                Object {
                  "fontWeight": "100",
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#e3e7ed",
            "height": 0.5,
          },
          undefined,
        ]
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "flex": 1,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingVertical": 20,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 9,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#484848",
                "fontWeight": "200",
              }
            }
          >
            Change Country
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#484848",
                  "fontSize": 20,
                },
                Object {
                  "fontWeight": "100",
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#e3e7ed",
            "height": 0.5,
          },
          undefined,
        ]
      }
    />
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flex": 1,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingVertical": 20,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 9,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#484848",
                  "fontWeight": "200",
                }
              }
            >
              My Properties
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#484848",
                    "fontSize": 20,
                  },
                  Object {
                    "fontWeight": "100",
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#e3e7ed",
              "height": 0.5,
            },
            undefined,
          ]
        }
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flex": 1,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingVertical": 20,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 9,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#484848",
                  "fontWeight": "200",
                }
              }
            >
              Logout
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#484848",
                    "fontSize": 20,
                  },
                  Object {
                    "fontWeight": "100",
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollContentView>
</RCTScrollView>
`;
